docs/src/charmil_config.md

```diff
@@ -0,0 +1,170 @@
+- Provides the host CLI developers with a set of methods to import plugin configuration maps and read/write configurations from/to a local config file.
+
+## Steps to Use:
+
+- ### For host CLI developers:
+
+  1.  Open the file where the root command of your CLI is defined using Cobra.
+
+  2.  Import the Charmil Config package by adding the following line at the top of that file:
+      ```go
+      import "github.com/aerogear/charmil/core/config"
+      ```
+  3.  Store a new instance of the Charmil Config handler by calling the `New` function.
+
+      _Example:_
+
+      ```go
+      h := config.New()
+      ```
+
+  4.  Create an instance of the `File` struct, passing all the required settings for the local config file as fields. Once this is done, pass that `File` instance as an argument into the `InitFile` method, in order to link the local config file to the pointer receiver ie. handler.
+
+      _Example:_
+
+      ```go
+      f = config.File{
+      	Name: "config",
+      	Type: "yaml",
+      	Path: "./examples/host",
+      }
+
+      h.InitFile(f)
+      ```
+
+  5.  Load config values from the local config file using the `Load` method.
+
+      _Example:_
+
+      ```go
+      err := h.Load()
+      if err != nil {
+      	log.Fatal(err)
+      }
+      ```
+
+  6.  Use the `SetValue` and `GetValue` methods respectively to set and get values to/from the current config.
+
+      _Example:_
+
+      ```go
+      h.SetValue("key0", "val0")
+
+      val, err := h.GetValue("key0") // returns "val0"
+      if err != nil {
+      	log.Fatal(err)
+      }
+      ```
+
+  7.  Map a plugin name to its config map using the `SetPluginCfg` method by passing the name of the plugin (as you want it in the config file) as the first argument and the config map imported from the plugin, as the second argument.
+
+      _Example:_
+
+      ```go
+      h.SetPluginCfg("pluginName", pluginCfg)
+      ```
+
+  8.  Call the `MergePluginCfg` method to store the config of every plugin (ie. mapped using the `SetPluginCfg` method) into the current config.
+
+      _Example:_
+
+      ```go
+      h.MergePluginCfg()
+      ```
+
+  9.  Write current config into the local config file using the `Save` method.
+
+      _Example:_
+
+      ```go
+      err = h.Save()
+      if err != nil {
+      	log.Fatal(err)
+      }
+      ```
+
+- ### For plugin developers:
+
+  1.  Open the file where the root command of your CLI is defined using Cobra.
+
+  2.  Import the Charmil Config package by adding the following line at the top of that file:
+      ```go
+      import "github.com/aerogear/charmil/core/config"
+      ```
+  3.  Store a new instance of the Charmil Config handler by calling the `New` function.
+
+      _Example:_
+
+      ```go
+      h := config.New()
+      ```
+
+  4.  Use the `SetValue` and `GetValue` methods respectively to set and get values to/from the current config.
```

Comment:
Imagine `rhoas kafka topics` commands becoming a plugin. This api will be huge downgrade and require refactor